verification.py

```diff
@@ -758,6 +759,271 @@ def test_sweep_ex1(self):
+    but in msprime only selection coefficient for AA can be specified, and
+    use h=0.5 to calculate that for Aa.
+    """
+
+    def _msms_str_to_parameters(self, msms_cmd):
+        """
+        parse msms cmdline arguments into a dictionary.  This method is called
+        by `_run_msp_sample_stats`
+
+        msms cmdline pattern:
+            nsam nrep -t theta -r rho num_sites -SF end_time end_frequency \
+            -SAA sAA -SaA saA -Sp sel_pos -N refsize -seed rand_seed
+
+        eg. "5 1 -t 200 -r 200 500000 -SF 0.002 0.9 -Sp 0.5"\
+            " -SaA 5000 -SAA 10000 -N 10000 -seed 1"
+        """
+        # intialize local variables
+        end_time_lst = []  # use list for multiple sweeps
+        end_frequency_lst = []
+        num_sweeps = 0
+        sAA = saA = sel_pos = -1.0
+        saA = -0.5
+        rand_seed = (random.randint(1, 2 ** 16),)
+
+        # parse arguments
+        tokens = msms_cmd.split(" ")
+        for ind in range(len(tokens)):
+            if ind == 0:
+                nsam = int(tokens[ind])
+                nrep = int(tokens[ind + 1])
+            elif tokens[ind] == "-t":
+                theta = float(tokens[ind + 1])
+            elif tokens[ind] == "-r":
+                rho = float(tokens[ind + 1])
+                num_sites = int(tokens[ind + 2]) + 1
+            elif tokens[ind] == "-SF":
+                num_sweeps += 1
+                end_time_lst.append(float(tokens[ind + 1]))
+                end_frequency_lst.append(float(tokens[ind + 2]))
+            elif tokens[ind] == "-Sp":
+                sel_pos = float(tokens[ind + 1])
+            elif tokens[ind] == "-SAA":
+                sAA = float(tokens[ind + 1])
+            elif tokens[ind] == "-SaA":
+                saA = float(tokens[ind + 1])
+            elif tokens[ind] == "-N":
+                refsize = int(tokens[ind + 1])
+            elif tokens[ind] == "-seed":
+                rand_seed = float(tokens[ind + 1])
+            else:
+                pass
+
+        # check if h = 0.5
+        if abs(saA * 2 - sAA) > 1e-5:
+            logging.warning(
+                "If 2 * saA is not equal to sAA, saA is set to sAA / 2,"
+                "that is h can only be 0.5 in msprime"
+            )
+            saA = sAA / 2.0
+
+        return {
+            "nsam": nsam,
+            "nrep": nrep,
+            "num_sweeps": num_sweeps,
+            "end_time_lst": end_time_lst,
+            "end_frequency_lst": end_frequency_lst,
+            "Ne": refsize,
+            "alpha": sAA,
+            "theta": theta,
+            "rho": rho,
+            "num_sites": num_sites,
+            "sel_pos": sel_pos,
+            "rand_seed": rand_seed,
+        }
+
+    def _msms_parms_to_run_msp(self, params):
+        """ run smulation for a single sample and return a tree sequence.
+            this method is called by `_run_msp_sample_stats` in a loop to
+            generate nrep samples.
+        """
+        Ne_temp = None
```

Comment:
I don't see what Ne_temp is for. I think it would be simpler to just set Ne=0.25 throughout, unless we're explicitly setting population sizes from the msms command line. It's confusing to have ``params["Ne"] and also this Ne parameter. is there a way we can simplify this?

Reply:
I know this is confusing. I was also confused too. For Hudson model, I need to use Ne= refsize, but for sweep model, I need to use Ne=0.25. I will re-organize this part to make it easier to see the difference.